### ISSUE TITLE
Finance : Input by Account

### ISSUE THREAD

**Issue by afs1jes** [afs1jes/CHNCT-Migration#22]

**Comment by afs1jes** [comment]
Hi Chrissy,

I want to make sure that I get the MustItemize settings correct on the input models. Can you confirm for me where (which accounts) you require itemizations?  It looks to me as though it's all Expense accounts that aren't calculated in the Comp model... does that sound right?

Also, can I assume that there are no LOB or Department exceptions to this rule?

Thanks

Jim

**Comment by stuartratner** [comment]
Error on Itemize - just shows Actuals columns, no Forecast. See attached model
Passed all other checks.

[ITEMIZE ERROR Revenue & Expense Input for 200 - Finance 01 - Corporate.xlsx](https://github.com/afs1jes/CHNCT-Migration/files/5315140/ITEMIZE.ERROR.Revenue.Expense.Input.for.200.-.Finance.01.-.Corporate.xlsx)

**Comment by rlautt** [comment]
Stu, to focus JIm's effort, can you diagnose the problem?

**Comment by afs1jes** [comment]
_transactions sheet modified the header so that the Period displays the queried period rather than the period of the first record returned.

**Comment by afs1jes** [comment]
Data loaded for 2020 Actuals and 2020 Working Forecast data created (not copied from prod, so there may not be a 100% match until data is re-copied. Itemization formatting looks correct.

**Comment by csoroka** [comment]
Hello Jim - I started to test the Input by Account Template for Finance 01-200 2021 Working Budget.  
I noticed that the 5012 account was shaded in blue and that I was able to type in amounts on the input tab - this account pulls from the compensation model and should not be updated by the user via this template via the input tab or through drill down. 
In the current system the 5010 & 5012 member rows are hidden and the users can only see the Salaries (parent) row.  Can you update the new system to display the same way?
I also was able to drill into 5010 and change the row amount for January.  The following rows should be protected so the user can not drill into them since the amounts are pulled from the compensation model:
5010 Salary
5012 Incentive Pool
5030 Payroll Taxes

Please let me know when these are updated and I will continue my testing.

Thank you,
Chrissy

**Comment by afs1jes** [comment]
I've added account 5012 - Incentive to the set of accounts that is formatted as locked, no fill. The complete list of accounts in this set include:
{
[5050 - PTO exp],
[5060 - PTO Payout],
[DSS Staffing],
[DSS Occupancy/Depreciation],
[Rewards to Quit Membership],
[5100 - Depreciation Exp],
[5010 - Salaries Expense],
[FDN 5010 - Salary],
[5012 - Incentive Pool],
[Consol_Salaries Expense],
[5030 - Payroll Taxes],
[FDN 5030 - Payroll Taxes]
}

In summary, 5012 is now locked from input in this template (and in the 5-yr version of this template).

**Comment by csoroka** [comment]
Thank you Jim for protecting the rows in the input template so I’m unable to enter dollars manually on that tab.
However, I’m able to drill into these rows that are protected and override what comes over from the compensation model.
See the $30,000 entered in Jan for 5010 and $2,000 for 5012 (LOB 01 & Dept 200)

Thank you,
Chrissy

**Comment by csoroka** [comment]
Hello Jim - I ran the input by account and have the following questions:

Drilling into Itemized data -
Can we hide the quarter columns - as the other itemized drill downs are displayed?
Is there a "remove row" option?  
I was able to drill into the 5010, 5012 and 5030 and change the amounts of the itemizations.  Since these amounts pull from the compensation model they should be protected.

Drilling into Actual data - 
The Date column isn't formatted correctly

Comp Data pulling into Input by Account - 
Tried to test again via the 2 year model and columns are blank that usually aren't (this data typically pulls from prior year data as a starting point).  I tried typing in the amounts under annual salary adding the FTEs to 2020 and putting in a hire date hoping that would populate something into the 2020 columns to test 2020/2021 and that didn't work.
Until that is fix I'm unable to test input by account.

Thank you,
Chrissy

**Comment by afs1jes** [comment]
The design of the new Itemization process kind of makes it hard to skip the Qtrs, but I'll see if I can come up with a workaround for that. 

There is no longer a need to click a button to remove a row. All you have to do is clear out the text in the "Itemization" column. That will remove the values on that row from the "Totals" at the top (which is the source for the link back to the Input sheet).  When you submit your model, these rows will be deleted from the database and should not return on a subsequent drill through.

I've updated template so that the locked rows (like the comp related accounts that get special formatting) are not allowed to be itemized. Instead, any drill to itemization from those rows will work just like calling itemizations from a rollup rows.... the sheet will open and will display any existing itemizations in a read-only layout (locked and with no link-back to the input sheet).

Your last time, related to "Comp Data pulling into Input by Account" seems like it's related to the Comp Input model and not this Finance Input by Account. Is that correct? Or am I not understanding your note?

**Comment by csoroka** [comment]
Hello Jim - I think prior you added the (+) and (-) buttons to the quarters.  Can those be added back?

As far as removing a row - I test it and it looks like all I need to remove is the Itemization column data (no other columns) and the entire row disappears is that correct? 
fyi - users utilize this model to earmark rows for the following year (renewals that only happen every other year or every 3 etc)  So they will type in information into the Itemization column but will not type in amounts in the monthly columns.  I just wanted to make sure that these type of entries will be carried over to other scenarios when copying.

In addition, it seems to take a bit of time to drill into itemized data via this template.  This didn't seem to be an issue prior.

The Date column is not formatted correctly when drill down into actuals via this model.

Thank you for adding the read-only displays for the itemized detail for the accounts pull from the comp model.

As far as the input by account and its relationship to the comp model, I wanted to double check that the input by account was pulling the correct data/dollars from the comp model.  To do this I went into the comp model, for Finance, to unhide the columns so I could cross check the amounts and then saw the data missing in the comp model.  That is why I had documented it here.

-Chrissy

**Comment by kclark-a3** [comment]
- Zero config itemizations
- template transactions

**Comment by kclark-a3** [comment]
Itemizations and Transactions style sheets added; this template is ready for testing.

**Comment by kclark-a3** [comment]
Received error below when tried submitting.

'Untrapped error occurred while processing your request. Application-defined or object-defined error.'

**Comment by csoroka** [comment]
I started to test this template for LOB 01 & Dept 200 and noticed the following issues:

- July-Dec for the following accounts on the input tab don't match the LID tab:
           -5130, 5200, 5210 & 5230
- Due to the fact that all users have access to this template we need to hide the following rolls and only show the rollup:
            - 5010 & 5012 
- If I highlight multiple cells and choose to pull up the LID for all those rolls at once, it only pull up one account tab at a time.  The current prod allows me to pull up multiple tabs, this is helpful for printing or for submitting

Thank you,
Chrissy

**Comment by csoroka** [comment]
What is the status on this template?

**Comment by kclark-a3** [comment]
I'll provide a status soon.

**Comment by kclark-a3** [comment]
Update

 - let's do a screenshare and discuss the LID issue at the next weekly meeting
 - accounts 5010 and 5012 are no longer visible
 - multi cell highlight is not available in .NET.
 -
[Revenue & Expense Input for_01 - Corporate_610 - IS Technical Svcs (1).xlsx](https://github.com/afs1jes/CHNCT-Migration/files/9049015/Revenue.Expense.Input.for_01.-.Corporate_610.-.IS.Technical.Svcs.1.xlsx)

**Comment by csoroka** [comment]
The LID now ties (I have attached the original that I had run when it didn't)
The multi cell highlight is used very frequently by our users - is there any way to program that in?

[Input by Account 01-200 2022 WF.pdf](https://github.com/afs1jes/CHNCT-Migration/files/9055705/Input.by.Account.01-200.2022.WF.pdf)

**Comment by AVHuntley** [comment]
Looks like multi-cell highlight is working now. When I highlight several adjacent itemizable cells in a column, then select **Itemize**, the Itemization (LID) sheet for each selected account is generated.

Test again to make sure?

**Comment by csoroka** [comment]
I have tested the multi-cell and it is working correctly.
The only remaining item I need to test with this template is that the numbers come over correctly from the comp model for salary.

-Chrissy

**Comment by csoroka** [comment]
Compensation model not carrying over correctly - see more detail under the 5yr input by account task.

**Comment by csoroka** [comment]
Please see recent findings for the 5 yr input model first.  When I went into the 2 year model I can see something I tried to enter itemized data into 5295 for dept 09-100 as a test that didn't carry forward but when I drill in the itemized line is gone but the $5 is there.  
I should not be able to drill into the "Salaries" line or the "5030" Taxes line on either input model, but I can.

**Comment by afs1jes** [comment]
A VBA module has been added that will check an account's ReadOnly status before opening an "Itemizations" sheet. If the account is "ReadOnly", the itemizations call is cancelled. If the account is not read only, Itemizations progresses as normal.  The only side-effect I've noticed is that, because rollup rows are marked as ReadOnly, those won't show itemizations in the input models.  If this is a problem, we could modify slightly... here are two ideas:  1) we could change the template rollup formatting so that rollup rows are not marked as ReadOnly.  2) We could set the "MustItemize"flag for restricted accounts to "-1" and chance the code to check for that. We'd need to make sure that this doesn't impact MustItemize.

**Comment by csoroka** [comment]
fyi - same issues that I posted regarding the 5 yr input template happened as well in the 2 yr input

**Comment by AVHuntley** [comment]
> Issues we discussed today on the phone:
> (1) When drilling into multiple LID rows I received an "Error 1004" pop up
> (2) Can the input tab automatically update after the "submit" button is clicked in the LID tabs?  Instead of clicking the "refresh" icon?
> (3) When I delete some of the "Itemization" descriptions the row totals all disappeared
> (4) When I deleted all the LID and submitted, the amounts still stayed in the input tab for the specific account (turned black font) the row didn't change to zeros across the board
> (5) When opening up the same account LID in two different years - we noticed the tab names need to reference the year and account 
> 
> Thank you,
> Chrissy

**Comment by AVHuntley** [comment]
Hey thanks for your patience with this most recent update. I believe we've addressed all 5 items noted in the last post. A few changes to the A3 add-in and to the templates' VBA seemed to resolve most issues.

I noticed a single issue that may never come up, but I figured I'd mention it - If you close an itemization tab _without_ using the on-sheet 'close' link, it can cause #REF errors that persist even after refreshing. Do you think this will be a problem, or is it unlikely to come up?

**Comment by csoroka** [comment]
I’m not sure that any of us know how to close an itemized tab without using the “close” link embedded in the sheet.  The only other way I know is to close the entire workbook and therefore wouldn’t see the #ref errors anyways.

**Comment by csoroka** [comment]
Looks good
